AndroidSDK/src/com/leanplum/internal/Request.java

```diff
@@ -383,11 +384,13 @@ private void sendNow() {
     Util.executeAsyncTask(true, new AsyncTask<Void, Void, Void>() {
       @Override
       protected Void doInBackground(Void... params) {
-        final List<Map<String, Object>> requestsToSend = getUnsentRequests();
+        List<Map<String, Object>> requestsToSend = getUnsentRequests();
         if (requestsToSend.isEmpty()) {
           return null;
         }
 
+        requestsToSend = updateUnsentRequests(requestsToSend);
```

Comment:
I agree, can the uuid be hold in shared preferences, until next send, and included during inserting of events....
only on sending the events, we will update the uuid?